src/schema.rs

```diff
@@ -844,7 +839,7 @@ fn pcf_map(schema: &Map<String, serde_json::Value>) -> String {
         }
 
         // Strip out unused fields ([STRIP] rule)
-        if field_ordering_position(k).is_none() {
+        if field_ordering_position(k).is_none() || k == "default" || k == "doc" || k == "aliases" {
```

Comment:
I know that the spec doesn't mention "default" as field to keep, but I think that changing the default value should actually make a schema different. What do you think?

Reply:
They mention `Strip all others (e.g., doc and aliases)` though. But beyond that, reading data that have been written with some default values isn't related to those values, only to their type. For instance those two schemas should be considered equal:

```
{
    "type": "record",
    "name": "test",
    "fields": [{"name": "a", "type": "long", "default": 42}]
}
```

and

```
{
    "type": "record",
    "name": "test",
    "fields": [{"name": "a", "type": "long", "default": 999}]
}
```

Reply:
It is even worse if one schema specifies `null` as a default value for a complex field (map, array) and another schema doesn't, because they would be exactly the same but their PCF would be different if we consider default value as part of it. And with the current implementation finding `null` (in the default value) just panics.